src/status_im/contexts/onboarding/events.cljs

```diff
@@ -1,31 +1,192 @@
 (ns status-im.contexts.onboarding.events
   (:require
     [quo.foundations.colors :as colors]
+    [re-frame.interceptor :as interceptor]
+    [react-native.mmkv :as mmkv]
     status-im.common.biometric.events
     [status-im.constants :as constants]
     [status-im.contexts.shell.constants :as shell.constants]
+    [status-im.feature-flags :as ff]
     [taoensso.timbre :as log]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]
     [utils.security.core :as security]))
 
+(defn notifications-setup-start
+  [{:keys [db]}
+   [{:keys [enable-notifications?
+            enable-news-notifications?
+            biometrics?
+            onboarding?
+            syncing?]}]]
+  {:db (if-not onboarding?
+         db
+         (update-in db
+                    [:onboarding/profile]
+                    assoc
+                    :enable-notifications?      enable-notifications?
+                    :enable-news-notifications? enable-news-notifications?))
+   :fx (cond-> []
+         (and (not onboarding?) enable-notifications? enable-news-notifications?)
+         (conj [:dispatch [:notifications/news-notifications-switch true]])
+
+         (and (not onboarding?) enable-notifications?)
+         (conj [:dispatch [:push-notifications/switch true]])
+
+         :always
+         (conj [:dispatch
+                [:onboarding/notifications-setup-done
+                 {:onboarding? onboarding?
+                  :syncing?    syncing?
+                  :biometrics? biometrics?}]]))})
+
+(rf/reg-event-fx :onboarding/notifications-setup-start notifications-setup-start)
+
+(defn notifications-setup-done
+  [{:keys [db]} [{:keys [biometrics? onboarding? syncing?]}]]
+  {:fx (cond-> []
+         (and biometrics? onboarding? syncing?)
+         (conj [:dispatch [:onboarding/finalize-setup]])
+
+         (and biometrics? onboarding? (not syncing?))
+         (conj [:dispatch [:onboarding/create-account-and-login]])
+
+         (and (not biometrics?) onboarding? syncing?)
+         (conj [:dispatch [:onboarding/finish-onboarding]])
+
+         (and (not biometrics?) onboarding? (not syncing?))
+         (conj [:dispatch [:onboarding/create-account-and-login]])
+
+         (and (not biometrics?) (not onboarding?) (not syncing?))
+         (conj [:dispatch [:shell/show-root-view]]))})
+
+
+(rf/reg-event-fx :onboarding/notifications-setup-done notifications-setup-done)
+
+(defn notifications-setup
+  [{:keys [db]}
+   [{:keys [biometrics-supported? biometrics? syncing? onboarding?]}]]
+  (let [db-view-id      (get-in db [:view-id])
+        current-view-id (if (= db-view-id :screen/onboarding.syncing-biometric)
+                          :screen/onboarding.enable-biometrics
+                          db-view-id)]
```

Comment:
Here we're handling a special case when attempting to navigate from the current view to the notification setup screen. During the onboarding flow with syncing, we've aliased to the onboarding view-id to `:screen/onboarding.syncing-biometric`, but that is not correct view-id for performing a navigation, so I replace it with the correct view-id